src/net/protocol_agnostic/stream.rs

```diff
@@ -370,3 +370,50 @@ where
         }
     }
 }
+
+#[cfg(test)]
+mod test {
+    use priv_prelude::*;
+    use config::DevConfigSettings;
+    use future_utils::bi_channel;
+    use tokio_core::reactor::Core;
+
+    #[test]
+    fn direct_connect_tcp_disabled() {
```

Comment:
It's also useful to spell out what's expected, e.g. 
```
#[test]
fn direct_connect_uses_other_than_tcp_addrs_when_tcp_disabled()
```
Personally, I like to scope tests:
```
mod direct_connect {
    #[test]
    fn it_uses_other_than_tcp_addrs_when_tcp_disabled {
    }
}
```